### ISSUE TITLE
Failed to deserialize InitializeParams: invalid type boolean, expected struct ChangeAnnotationWorkspaceEditClientCapabilities

### ISSUE THREAD

**Issue by jplatte** [rust-analyzer/rust-analyzer#7078]
When trying to use rust-analyzer on [Ruma](https://github.com/ruma/ruma), it fails to initialize.

ra version: 20201228 (pkgver 1, from the arch linux community repo)
environment: VSCodium with `matklad.rust-analyzer` v0.2.416
full error message:

```plain
Failed to deserialize InitializeParams: capabilities.workspace.workspaceEdit.changeAnnotationSupport: invalid type: boolean `true`, expected struct ChangeAnnotationWorkspaceEditClientCapabilities; {"processId":7313,"clientInfo":{"name":"vscode","version":"1.52.1"},"locale":"de","rootPath":"/home/jplatte/Projekte/ruma","rootUri":"file:///home/jplatte/Projekte/ruma","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional","normalizesLineEndings":true,"changeAnnotationSupport":true},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"tagSupport":{"valueSet":[1]}},"codeLens":{"refreshSupport":true},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true,"semanticTokens":{"refreshSupport":true}},"textDocument":{"publishDiagnostics":{"relatedInformation":true,"versionSupport":false,"tagSupport":{"valueSet":[1,2]},"codeDescriptionSupport":true,"dataSupport":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true,"tagSupport":{"valueSet":[1]},"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"insertTextModeSupport":{"valueSet":[1,2]}},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true},"activeParameterSupport":true},"contextSupport":true},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true,"tagSupport":{"valueSet":[1]},"labelSupport":true},"codeAction":{"dynamicRegistration":true,"isPreferredSupport":true,"disabledSupport":true,"dataSupport":true,"resolveSupport":{"properties":["edit"]},"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"honorsChangeAnnotations":false},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true,"prepareSupportDefaultBehavior":true,"honorsChangeAnnotations":true},"documentLink":{"dynamicRegistration":true,"tooltipSupport":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true},"selectionRange":{"dynamicRegistration":true},"callHierarchy":{"dynamicRegistration":true},"semanticTokens":{"dynamicRegistration":true,"tokenTypes":["namespace","type","class","enum","interface","struct","typeParameter","parameter","variable","property","enumMember","event","function","member","macro","keyword","modifier","comment","string","number","regexp","operator"],"tokenModifiers":["declaration","definition","readonly","static","deprecated","abstract","async","modification","documentation","defaultLibrary"],"formats":["relative"],"requests":{"range":true,"full":{"delta":true}},"multilineTokenSupport":false,"overlappingTokenSupport":false},"onTypeRename":{"dynamicRegistration":true}},"window":{"showMessage":{"messageActionItem":{"additionalPropertiesSupport":true}},"showDocument":{"support":true},"workDoneProgress":true},"general":{"regularExpressions":{"engine":"ECMAScript","version":"ES2020"},"markdown":{"parser":"marked","version":"0.6.2"}},"experimental":{"snippetTextEdit":true,"codeActionGroup":true,"hoverActions":true,"statusNotification":true}},"initializationOptions":{"cargoRunner":null,"runnableEnv":null,"inlayHints":{"enable":false,"chainingHints":true,"maxLength":null,"parameterHints":true,"typeHints":true},"updates":{"channel":"stable","askBeforeDownload":true},"serverPath":"/usr/bin/rust-analyzer","trace":{"server":"off","extension":false},"debug":{"engine":"auto","sourceFileMap":{"/rustc/<id>":"${env:USERPROFILE}/.rustup/toolchains/<toolchain-id>/lib/rustlib/src/rust"},"openDebugPane":false,"engineSettings":{}},"assist":{"importMergeBehaviour":"full","importPrefix":"plain"},"callInfo":{"full":false},"cargo":{"autoreload":true,"allFeatures":false,"features":[],"loadOutDirsFromCheck":true,"noDefaultFeatures":false,"target":null,"noSysroot":false},"checkOnSave":{"enable":true,"allFeatures":null,"allTargets":true,"command":"clippy","noDefaultFeatures":null,"target":null,"extraArgs":[],"features":null,"overrideCommand":null},"completion":{"addCallArgumentSnippets":true,"addCallParenthesis":true,"postfix":{"enable":true},"autoimport":{"enable":true}},"diagnostics":{"enable":true,"enableExperimental":true,"disabled":[],"warningsAsHint":[],"warningsAsInfo":[]},"files":{"watcher":"client"},"hoverActions":{"debug":true,"enable":true,"gotoTypeDef":true,"implementations":true,"run":true,"linksInHover":true},"lens":{"debug":true,"enable":true,"implementations":true,"run":true,"methodReferences":false},"linkedProjects":[],"lruCapacity":null,"notifications":{"cargoTomlNotFound":true},"procMacro":{"enable":true},"runnables":{"overrideCargo":null,"cargoExtraArgs":[]},"rustcSource":null,"rustfmt":{"extraArgs":[],"overrideCommand":null}},"trace":"off","workspaceFolders":[{"uri":"file:///home/jplatte/Projekte/ruma","name":"ruma"}]}
```

**Comment by lnicola** [comment]
Does it work with a newer version of the extension?

**Comment by jplatte** [comment]
That's the latest version available [on open-vsx](https://open-vsx.org/extension/matklad/rust-analyzer).

**Comment by jplatte** [comment]
Okay, apparently this is fixed in 0.2.432, I hope open-vsx will update soon.
